src/test/java/seedu/mtracker/console/AddCryptoParserTest.java

```diff
@@ -0,0 +1,85 @@
+package seedu.mtracker.console;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class AddCryptoParserTest {
+    public static final int PARAMETER_SIZE = 5;
+
+    public static String USER_INPUT_NO_EXPIRY = "TestName%1$s23.4%1$spositive%1$s%1$sTestRemarks";
+    public static String[] EXPECTED_PARAMS_NO_EXPIRY = {"TestName", "23.4", "positive", "", "TestRemarks"};
+
+    public static String USER_INPUT_NO_REMARKS = "TestName%1$s23.4%1$spositive%1$s18Oct%1$s ";
+    public static String[] EXPECTED_PARAMS_NO_REMARKS = {"TestName", "23.4", "positive", "18Oct", ""};
+
+    public static String USER_INPUT_WITH_REMARKS_AND_EXPIRY = "TestName%1$s100.4%1$snegative%1$s18Oct%1$sTestRemarks";
+    public static String[] EXPECTED_PARAMS_WITH_REMARKS_AND_EXPIRY = {"TestName", "100.4", "negative",
+        "18Oct", "TestRemarks"};
+    public static String USER_INPUT_TRY_INVALID_NAME = "%1$s%1$s%1$sTestName%1$s23.4%1$spositive%1$s18Oct%1$s ";
+    public static String USER_INPUT_TRY_INVALID_PRICE = "%1$sTestName%1$s2sd3.4%1$s100.4"
+            + "%1$snegative%1$s18Oct%1$sTestRemarks";
+    public static String USER_INPUT_TRY_INVALID_SENTIMENT = "%1$sTestName%1$s100.4"
+            + "%1$swrong%1$s%1$snegative%1$s18Oct%1$sTestRemarks";
```

Comment:
My bad the variables should have final I missed out on the stockparser test, can help me add in also